### PR TITLE
Seed SojournBroker.Worker's random number generator for its backoff.

### PR DESCRIPTION
**Pull request by MSch**
/cc @fishcakez

### CODE REVIEW

File: lib/ecto/pools/sojourn_broker/worker.ex [hunk 1]
```diff
@@ -41,6 +41,9 @@ defmodule Ecto.Pools.SojournBroker.Worker do
     backoff_threshold = div(max_backoff, 3)
     lazy = Keyword.get(worker_opts, :lazy, true)
 
+    # Seed random number generator for backoff
+    :random.seed(:erlang.now)
```
Comment (lexmag) [inline lib/ecto/pools/sojourn_broker/worker.ex]: Could it be `:os.timestamp()`?

Comment (fishcakez) [inline lib/ecto/pools/sojourn_broker/worker.ex]: `:erlang.now/0` will generate a warning in 18.0+ so we shouldn't introduce it.

This 18.0 docs for `:random` suggest:
```erlang
random:seed(erlang:phash2([node()]), erlang:monotonic_time(), erlang:unique_integer())
```

I guess we could use the has of a reference for a unique enough number:
```elixir
{_, Sec, Micro} = :os.timestamp()
{Sec, Micro, :erlang.phash2([make_ref()])}
```

Comment (josevalim) [inline lib/ecto/pools/sojourn_broker/worker.ex]: Are we relying on the different workers having different seeds? If not, `:os.timestamp` is fine. If you need something else, maybe even the last example with phash2 of `self()`?

Comment (fishcakez) [inline lib/ecto/pools/sojourn_broker/worker.ex]: Yes we want all the workers to have different seeds.

Comment (fishcakez) [inline lib/ecto/pools/sojourn_broker/worker.ex]: The randomess is so that the workers dont backoff in unison.

Comment (MSch) [inline lib/ecto/pools/sojourn_broker/worker.ex]: As an aside: The actual implementation in rand.erl doesn't use `erlang:phash2([node()])` it uses `erlang:phash2([{node(),self()}])`.

https://github.com/erlang/otp/blob/maint/lib/stdlib/src/rand.erl#L88-L90

Comment (fishcakez) [inline lib/ecto/pools/sojourn_broker/worker.ex]: Lets copy `:rand` and use that hash.

**Pull request merged**